components/omega/doc/design/EOS.md

```diff
@@ -0,0 +1,158 @@
+(omega-design-eos)=
+
+# Equation of State
+
+## 1 Overview The equation of state relates density to the prognostic state
+variables temperature and salinity.  The prognostic temperature, $\theta$, is
+either potential temperature or Conservative Temperature, depending on the
+chosen equation of state, and  $S$ is the salinity.  Given  that the governing
+equations are  non-Boussinesq, the equation of state class or functor will
+provide specific volume from the state variables.  It will provide methods for
+computing the specific volume, and its first derivatives.
+
+## 2 Requirements
+
+### 2.1 Requirement: calculate specific volume/density
+The specific volume and density will be calculated form from prognostic
+temperature, salinity, and pressure (ref function `gsw_specvol(sa,ct,p`)).
+
+### 2.2 Requirement: allow a choice of equations of state
+Omega-1 should include the option for two EOS options. The first is a linear
+eos, for benchmarking against MPAS-O and to use in simplified cases. The other
+option should be a well-established version of TEOS-10 (e.g. Roquet 75-term
+polynomial). 
+
+### 2.3 Requirement: computational efficiency
+The eos code should not be major computational bottleneck (criteria?)
```

Comment:
Bottleneck is unrealistic. In every ocean model I've worked with, the EOS is one of the 2-3 top kernels and is always a bottleneck. TEOS-10 is even more expensive with a larger polynomial. Maybe something more like: As one of the most expensive kernels, the EOS must be as efficient as possible. Developers should be discouraged from calling the EOS too frequently.

Reply:
Agreed. As you can tell, I was trying to convey *an* expectation of reasonable performance, but struggled to find the right wording/criteria. I appreciate your suggestion.